lib/diagnostic.rb

```diff
@@ -43,6 +51,9 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    filename.each_with_object(new_string: '') do |line|
```

Comment:
`filename` is just like it sounds - the name of the file. It is not the file itself.